docs/diagrams/FindCoursesCommand.puml

```diff
@@ -0,0 +1,69 @@
+@startuml
+
+hide footbox
+
+!include .SequenceDiagram.config
+
+actor CEG_Student
+
+participant ":FindCoursesCommand" as FindCoursesCommand
+participant ":Storage" as Storage
+participant ":UI" as UI
+participant ":Course" as Course
+
+CEG_Student -> FindCoursesCommand: execute(userInput:String, storage:Storage)
+activate FindCoursesCommand
+
+FindCoursesCommand -> FindCoursesCommand: getKeyword(userInput:String)
+activate FindCoursesCommand
+opt keyword is null
+    FindCoursesCommand -> FindCoursesCommand: throw IllegalArgumentException
+end
+return keyword:String
+
+FindCoursesCommand -> FindCoursesCommand: findCommand(keyword:String)
+activate FindCoursesCommand
+FindCoursesCommand -> Storage: loadAllCourses()
+activate Storage
+return mappedCourses: List<Course>
+
+opt mappedCourses is empty
+FindCoursesCommand -> UI: printEmptyList()
+activate UI
+return
+end
+
+FindCoursesCommand -> FindCoursesCommand: matchKeyword(...)
+activate FindCoursesCommand
+loop For all course in mappedCourses
+FindCoursesCommand -> FindCoursesCommand: getMappedCode(course:Course)
+activate FindCoursesCommand
+FindCoursesCommand -> Course: getNusCourseCode()
+activate Course
+return nusCourseCode: String
+return nusCourseCode: String
+
+opt nusCourseCode equals to keyword
+FindCoursesCommand -> FindCoursesCommand: add course to foundCourses:List<Course>
+end
+end
+return
+
+FindCoursesCommand -> FindCoursesCommand: printFindCommand(foundCourses: List<Course>)
+activate FindCoursesCommand
+alt foundCourses is empty
+    FindCoursesCommand -> FindCoursesCommand: throw IllegalArgumentException
+else else
+    loop For all course in foundCourses
+        FindCoursesCommand -> UI: printFoundCourses(...)
+        activate UI
+        return
+    end
+end
+return
```

Comment:
![image](https://github.com/user-attachments/assets/b711495e-7a0a-4694-8b8a-f6f8b02c1c76)

![image](https://github.com/user-attachments/assets/511f0e9c-7e05-492f-853a-d07eda649c5a)

i think print is not a self-invocation

throw: though there is no standard notation, it may be better to show the throw as a call to exception object rather than self invocation

Otherwise the rest LGTM!